static/js/pages/SitesDashboard.tsx

```diff
@@ -29,51 +29,67 @@ export function siteDescription(site: Website): string | null {
   return null
 }
 
-export default function SitesDashboard(): JSX.Element {
-  const { search, pathname } = useLocation()
-  const history = useHistory()
+function getListingParams(search: string): WebsiteListingParams {
   const qsParams = new URLSearchParams(search)
-
   const offset = Number(qsParams.get("offset") ?? 0)
   const searchString = qsParams.get("q")
 
-  const listingParams: WebsiteListingParams = searchString ?
-    {
-      offset,
-      search: searchString
-    } :
-    {
-      offset
-    }
+  return searchString ? { offset, search: searchString } : { offset }
+}
 
-  const [{ isPending }] = useRequest(websiteListingRequest(listingParams))
+export default function SitesDashboard(): JSX.Element {
+  const { search } = useLocation()
+  const history = useHistory()
 
-  const listing: WebsiteListingResponse = useSelector(getWebsiteListingCursor)(
-    offset
+  const [listingParams, setListingParams] = useState<WebsiteListingParams>(() =>
+    getListingParams(search)
   )
 
-  const [searchInput, setSearchInput] = useTextInputState()
+  const [searchInput, setSearchInput] = useTextInputState(
+    listingParams.search ?? ""
+  )
 
+  /**
+   * This debounced effect listens on the search input and, when it is
+   * different from the value current set on `listingParams`, will format a new
+   * query string (with offset reset to zero) and push that onto the history
+   * stack.
+   *
+   * We are using the URL and the browser's history mechanism as our source of
+   * truth for when we are going to re-run the search and whatnot. So in this
+   * call we're just concerned with debouncing user input (on the text input)
+   * and then basically echoing it up to the URL bar every so often. Below we
+   * listen to the `search` param and regenerate `listingParams` when it
+   * changes.
+   */
   useDebouncedEffect(
     () => {
-      const currentSeach = searchString ?? ""
-      if (searchInput !== currentSeach) {
+      const currentSearch = listingParams.search ?? ""
+      if (searchInput !== currentSearch) {
         const newParams = new URLSearchParams()
-        newParams.set("offset", String(offset))
         newParams.set("q", searchInput)
```

Comment:
I noticed that if you delete the search text, the url ends in `sites/?offset=0&q=`.

I can't imagine that the empty string `q` would ever matter? But we could do `if (searchInput)` here if avoiding it is desirable.

Reply:
mmm that's definitely a little bit cleaner that way, I'll make that change